### PR TITLE
Add build metadata for Poetry/PyPI

### PR DESCRIPTION
**Pull request by nimasmi**
This adds to the `[tool.poetry]` section of pyproject.toml as per https://python-poetry.org/docs/pyproject/.

~~However, on testing, the metadata doesn't seem to be parsed by PyPI. https://test.pypi.org/project/django-gov-notify/0.1.1/#description.~~

~~PyPI's relevant documentation only covers setuptools, https://packaging.python.org/tutorials/packaging-projects/#configuring-metadata, so I'm not sure whether this is the right way to do it.~~

It was. I hadn't built the wheel before distributing, but had an old test wheel from a previous deployment to test PyPI. The metadata now shows at https://test.pypi.org/project/django-gov-notify/0.1.2/.

### CODE REVIEW

**Comment by zerolab** [comment]
Not sure why it didn't work.
I've just tested this with wagtail-markdown - https://test.pypi.org/project/wagtailmarkdown/

<details>
<summary>pyproject.toml</summary>
```toml
[tool.poetry]
name = "wagtailmarkdown"
version = "0.7.0-rc1"
description = "Markdown support for Wagtail"
authors = ["Dan Braghis <dan.braghis@torchbox.com>"]
readme = "README.md"
homepage = "https://github.com/torchbox/wagtail-markdown"
repository = "https://github.com/torchbox/wagtail-markdown"
keywords = ["Wagtail", "Django", "Markdown"]

[tool.poetry.dependencies]
Markdown = ">=3,<4"
bleach = ">=3.3.0,<4"
wagtail = ">=2.0"

[tool.poetry.dev-dependencies]
black = "^20.8b1"
isort = "^5.8.0"
flake8 = "^3.9.1"
pre-commit = "^3.4.0"

[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"

[tool.black]
target-version = ['py36']

[tool.isort]
profile = "black"
known_django = "django"
known_wagtail = "wagtail"
known_first_party = ['wagtailmarkdown']
sections = ['FUTURE', 'STDLIB', 'DJANGO', 'WAGTAIL', 'THIRDPARTY', 'FIRSTPARTY', 'LOCALFOLDER']
skip=['migrations', '.git', '__pycache__', 'venv']
```
</details>

then

```sh
$ poetry build
$ poetry config repositories.testpypi https://test.pypi.org/legacy/
$ poetry publish -r testpypi
Username: *edited*
Password:
Publishing wagtailmarkdown (0.7.0-rc1) to testpypi
 - Uploading wagtailmarkdown-0.7.0rc1-py2.py3-none-any.whl 100%
 - Uploading wagtailmarkdown-0.7.0rc1.tar.gz 100%
```

File: pyproject.toml [hunk 1]
```diff
@@ -1,8 +1,12 @@
 [tool.poetry]
 name = "django-gov-notify"
 version = "0.2.0"
-description = ""
+description = "A GOV.UK Notify flavoured Django email backend"
 authors = ["Nick Smith <nick.smith@torchbox.com>"]
+readme = "README.md"
+homepage = "https://github.com/nimasmi/django-gov-notify"
+repository = "https://github.com/nimasmi/django-gov-notify"
+keywords = ["email", "Django"]
```
Comment (zerolab) [inline pyproject.toml:9]: How about expanding this a bit to include classifiers and license?

```suggestion
keywords = ["email", "Django"]
classifiers = [
    "Development Status :: 4 - Beta",
    "Environment :: Web Environment",
    "Intended Audience :: Developers",
    "Operating System :: OS Independent",
    "Programming Language :: Python",
    "Programming Language :: Python :: 3",
    "Topic :: Communications :: Email",
    "Framework :: Django",
    "Framework :: Django :: 2.2",
    "Framework :: Django :: 3.1",
    "Framework :: Django :: 3.2",
]
license = BSD-2-Clause"
```

**Comment by zerolab** [comment]
Looks great!
Made a minor, non-blocking suggestion.

**Comment by nimasmi** [comment]
> Not sure why it didn't work.

Did you see my description update above? It was because I had an existing 0.1.1 release, that I'd made for test PyPI, and I forgot to build a second time.

The Wagtail Markdown metadata looks good.

**Comment by zerolab** [comment]
> Did you see my description update above? It was because I had an existing 0.1.1 release, that I'd made for test PyPI, and I forgot to build a second time.

Just saw it. Your edit and my comment must've crossed wires. Glad it was not some esoteric packaging issue

**Comment by nimasmi** [comment]
> > Did you see my description update above? It was because I had an existing 0.1.1 release, that I'd made for test PyPI, and I forgot to build a second time.
> 
> Just saw it. Your edit and my comment must've crossed wires. Glad it was not some esoteric packaging issue

PEBKAC

**Comment by zerolab** [comment]
on d116f3d and python classifiers -- had them initially, but removed because of this comment from https://python-poetry.org/docs/pyproject/#classifiers

> Note that Python classifiers are still automatically added for you and are determined by your python requirement.

**Comment by nimasmi** [comment]
> on [d116f3d](https://github.com/nimasmi/django-gov-notify/commit/d116f3d30e75a7dbe17fd927746d17e745252a22) and the Python classifiers -- had them initially, but removed because of this comment from https://python-poetry.org/docs/pyproject/#classifiers
> 
> > Note that Python classifiers are still automatically added for you and are determined by your python requirement.

Hmm. I went straight to PyPI, so didn't see that. That's confusing, because at https://test.pypi.org/project/django-gov-notify/, Pythons 3, 3.7, and 3.8 are listed, but not 3.6.

**Comment by zerolab** [comment]
Being explicit is better than relying on internal poetry voodoo, so +💯 for this PR to go in

**Comment by nimasmi** [comment]
Curious. On Test PyPI:

- https://test.pypi.org/project/django-gov-notify/0.1.4/, no Python type classifiers, lists 3, 3.7, 3.8
- https://test.pypi.org/project/django-gov-notify/0.1.3/, with Python type classifiers, lists 3, 3.6, 3.7, 3.8

**Pull request merged**